ui/page/main_page.go

```diff
@@ -400,14 +400,20 @@ func (mp *MainPage) HandleUserInteractions() {
 				if mp.receivePage == nil {
 					mp.receivePage = NewReceivePage(mp.Load)
 				}
-				pg = mp.receivePage
+				pg = NewReceivePage(mp.Load)
```

Comment:
Generating a new instance of the page every time a user visits the receive page is going to produce an unpredictable back-stack. Also if a user should navigate back to the previously generated page on the back-stack they'll have the old receiving address which sort of defeats your purpose.
What you should do is withing the receiving page `OnNavigatedTo()` retrieve the current selected wallet and call `wallet.CurrentAddress()` on the wallet, then assign the result to the page's `currentAddress`.